crates/emitter/src/control_structures.rs

```diff
@@ -122,7 +123,9 @@ pub trait Continuable {
     fn emit_continue_target_and_patch(&mut self, emit: &mut InstructionWriter);
 }
 
+#[derive(Debug, PartialEq)]
 pub struct LoopControl {
+    scope_note_index: ScopeNoteIndex,
```

Comment:
loop control itself doesn't have any scope.
so this field seems to be wrong,
or maybe it needs better name what it points.